### PR TITLE
PACA-265: RDA API 0.5 updates

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[PACA-265](https://jira.cms.gov/browse/PACA-265)

**User Story or Bug Summary:**

Update RDA API Client to Consume New Data from v0.5.

---

### What Does This PR Do?
Updates RDA API Client to Consume New Data from v0.5

- Updates all RDA API proto files with appropriate versions for RDA API 0.5. **(These files are copied directly from the RDA API [release artifacts](https://github.cms.gov/medicare-payment-modernization/rda-api).) so they don't require review except for context on what the API is returning to us.**
- Adds new tables to the `pre_adj` schema:
  - FissAuditTrails
  - McsAdjustments
  - McsAudits
  - McsLocations
- Adds a new database migration script `V37__preadj_rda_0_5_table_updates.sql`
- Updates `FissClaimTransformer` and `McsClaimTransformer` to support the new claim data messages returned by RDA API 0.5. **Note: These classes now use code generated from a DSL file.  Using the generator reduced development time substantially for the PR.  This led to some LOC changes but the functionality works the same way as before.**
- Updates `RandomFissClaimGenerator` and `RandomMcsClaimGenerator` to produce the new claim data messages.
- Expands the unit tests for `FissClaimTransformer` and `McsClaimTransformer` to provide more robust set of tests for each field.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:
* Verify all PR security questions and checklists have been completed and addressed.
* Verify that all new fields are present and tested.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

Since this PR contains a database migration we will need to merge and deploy [PR 861](https://github.com/CMSgov/beneficiary-fhir-data/pull/861) to disable hibernate validation prior to merging this PR.

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsDiagnosisCode.java [hunk 1]
```diff
@@ -24,34 +24,35 @@
...
-  @Column(name = "`priority`", nullable = false)
   @EqualsAndHashCode.Include
+  @Column(name = "`priority`", nullable = false)
   private short priority;
 
   @Column(name = "`idrDiagIcdType`", length = 1)
   private String idrDiagIcdType;
 
-  @Column(name = "`idrDiagCode`", length = 7)
+  @Column(name = "`idrDiagCode`", nullable = false, length = 7)
```
Comment (ericbuckley) [inline apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsDiagnosisCode.java:43]: 🎉

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissDiagnosisCode.java [hunk 2]
```diff
@@ -24,23 +24,23 @@
...
   @EqualsAndHashCode.Include
+  @Column(name = "`priority`", nullable = false)
   private short priority;
 
-  @Column(name = "`diagCd2`", length = 7, nullable = false)
+  @Column(name = "`diagCd2`", nullable = false, length = 7)
   private String diagCd2;
 
-  @Column(name = "`diagPoaInd`", length = 1, nullable = false)
+  @Column(name = "`diagPoaInd`", nullable = false, length = 1)
```
Comment (ericbuckley) [inline apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissDiagnosisCode.java:43]: I thought the RDA API team changed this to be nullable?  https://confluenceent.cms.gov/pages/viewpage.action?pageId=233814218

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V37__preadj_rda_0_5_table_updates.sql [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbPickupAddres2" varchar(20);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbPickupCity" varchar(20);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbPickupState" varchar(2);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbPickupZipcode" varchar(9);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbDropoffName" varchar(24);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbDropoffAddrL1" varchar(25);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbDropoffAddrL2" varchar(20);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbDropoffCity" varchar(20);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbDropoffState" varchar(2);
+ALTER TABLE "pre_adj"."McsDetails" ADD "idrDtlAmbDropoffZipcode" varchar(9);
```
Comment (ericbuckley) [inline apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V37__preadj_rda_0_5_table_updates.sql:170]: DO we need an alter statement to update the MCS diagnosis code table `idr_diag_code` value to be not null?

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/proto/rda/VERSION.txt [hunk 4]
```diff
@@ -1 +1 @@
-0.4.0
+0.5.0
```
Comment (ericbuckley) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/proto/rda/VERSION.txt:1]: will be 0.5.1 later today....

**Comment by ericbuckley** [comment]
@brianburton what do you think about adding a comment in the PR description to remind the BFD team that all of the "proto" file changes are just being copied over from https://github.cms.gov/medicare-payment-modernization/rda-api?  It seems like about 50% of the file changes in this PR are just updating those references.

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissDiagnosisCode.java (hunk 2 above)
Comment (brianburton) [reply apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissDiagnosisCode.java:43]: Great catch!

File: apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V37__preadj_rda_0_5_table_updates.sql (hunk 3 above)
Comment (brianburton) [reply apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V37__preadj_rda_0_5_table_updates.sql:170]: Thanks.  Will add that.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/proto/rda/VERSION.txt (hunk 4 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/proto/rda/VERSION.txt]: Updated.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/RdaLoadJobIT.java [hunk 5]
```diff
@@ -125,6 +148,26 @@ public void invalidFissClaimTest() throws Exception {
...
   }
 
+  /**
+   * All of our test claims should be valid for our IT tests to succeed. This test ensures this is
+   * the case and catches any incompatibility issues when a new RDA API version contains breaking
+   * changes.
+   */
+  @Test
+  public void mcsClaimsAreValid() throws Exception {
+    final ImmutableList<McsClaimChange> expectedClaims =
```
Comment (KyRobbins) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/RdaLoadJobIT.java:158]: I've noticed a number of variables declared as `ImmutableList` (or similar data structure) in the code in various places I believe.  Is there a specific need for this?  Typically you'd want to declare the interface `List` so that the code isn't quite as restricting.  For instance, later we might want to update all this logic to use JDK11 features like `List.of()` instead of `ImmutableList.of()` if it seemed appropriate.

**Review by KyRobbins — COMMENTED** [review]
Couple questions and notes

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/RdaLoadJobIT.java (hunk 5 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/RdaLoadJobIT.java:158]: I agree but ImmutableList has declarative value since it makes it perfectly clear that this is an immutable field.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformerTest.java [hunk 6]
```diff
@@ -434,598 +482,1233 @@ public void requiredFieldsMissing() {
...
+  }
+
+  @Test
+  public void testClaimHicNo() {
+    new ClaimFieldTester()
+        .stringField(FissClaim.Builder::setHicNo, PreAdjFissClaim::getHicNo, "hicNo", 12);
+  }
+
+  @Test
+  public void testClaimCurrStatus() {
```
[the remark this answers is not in the document]
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformerTest.java:501]: I have renamed the ClaimTransformerFieldTester methods to make their purpose clearer.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc-server/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/StoreRdaJsonApp.java [hunk 7]
```diff
@@ -68,21 +68,23 @@ public static void main(String[] args) throws Exception {
     }
   }
 
-  private static ManagedChannel createChannel(String host, int port) {
-    final ManagedChannelBuilder<?> channelBuilder = ManagedChannelBuilder.forAddress(host, port);
-    if (host.equals("localhost")) {
+  private static ManagedChannel createChannel(Config config) {
+    final ManagedChannelBuilder<?> channelBuilder =
+        ManagedChannelBuilder.forAddress(config.grpcConfig.getHost(), config.grpcConfig.getPort());
+    if (config.grpcConfig.getHost().equals("localhost")) {
```
Comment (ericbuckley) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc-server/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/StoreRdaJsonApp.java:74]: what about 127.0.0.1?  Maybe `InetAddress.getByName(config.grpcConfig.getHost()).isAnyLocalAddress()`?

**Review by ericbuckley — APPROVED** [review]
👍

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc-server/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/StoreRdaJsonApp.java (hunk 7 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc-server/src/main/java/gov/cms/bfd/pipeline/rda/grpc/sink/StoreRdaJsonApp.java:74]: That's a good enhancement that we could apply in a few places in a later PR.

**Review by KyRobbins — DISMISSED** [review]

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissClaim.java [hunk 8]
```diff
@@ -141,6 +134,158 @@
...
+  /** Adjustment Requestor Identification */
+  @Column(name = "`adjReqCd`", length = 1)
+  private String adjReqCd;
+
+  /** Adjustment Reason Code */
+  @Column(name = "`adjReasCd`", length = 2)
+  private String adjReasCd;
+
+  /** Cancel Cross-reference Document Control Number */
+  @Column(name = "`cancelXrefDcn`", length = 23)
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissClaim.java:162]: Thank you for these comments; some of these names are really inscrutable!

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissDiagnosisCode.java [hunk 9]
```diff
@@ -49,12 +50,13 @@
   @Column(name = "`lastUpdated`")
   private Instant lastUpdated;
 
+  /** PK class for the FissDiagnosisCodes table */
   @Data
   @NoArgsConstructor
   @AllArgsConstructor
-  /* PK class for the FissDiagnosisCodes table */
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissDiagnosisCode.java:55]: I had never thought much about it, but after some research putting the documentation before the annotations is the recommended way to document. TIL! 🎉

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissPayer.java [hunk 10]
```diff
@@ -75,50 +70,50 @@
...
   @Column(name = "`insuredGroupName`", length = 17)
   private String insuredGroupName;
 
-  // Insured only
+  /** Insured only */
   @Column(name = "`insuredGroupNbr`", length = 20)
   private String insuredGroupNbr;
 
-  // BeneZ only
+  /** BeneZ only */
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissPayer.java:112]: ❓ Wondering if there is a better way to indicate that some fields are only for some PayerTypes, in code. Maybe multiple classes that extend a base class? Not sure if that'd work with the implementation here, but just a thought.

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsAdjustment.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+  @Id
+  @Column(name = "`priority`", nullable = false)
+  @EqualsAndHashCode.Include
+  private short priority;
+
+  @Column(name = "`lastUpdated`")
+  private Instant lastUpdated;
+
+  @Column(name = "`idrAdjDate`")
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsAdjustment.java:46]: 🍰  Could we get those handy comments about what these fields names mean in this class too?

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsDetail.java [hunk 12]
```diff
@@ -95,15 +96,70 @@
...
+  @Column(name = "`idrDtlRendSpec`", length = 2)
+  private String idrDtlRendSpec;
+
+  @Column(name = "`idrDtlRendNpi`", length = 10)
+  private String idrDtlRendNpi;
+
+  @Column(name = "`idrDtlRendProv`", length = 10)
+  private String idrDtlRendProv;
+
+  @Column(name = "`idrKDtlFacProvNpi`", length = 10)
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsDetail.java:117]: 🍰  Field name comments might help on column names like this one too!

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGeneratorTest.java [hunk 13]
```diff
@@ -21,31 +21,48 @@ public void randomClaim() throws InvalidProtocolBufferException {
         "{\n"
             + "  \"dcn\": \"9845557\",\n"
             + "  \"hicNo\": \"904843533707\",\n"
-            + "  \"currStatusEnum\": \"CLAIM_STATUS_MOVE\",\n"
+            + "  \"currStatusEnum\": \"CLAIM_STATUS_ROUTING\",\n"
```
[the remark this answers is not in the document]
Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomFissClaimGeneratorTest.java:24]: This would be pretty trivial to load from a file, a huge chunk of json should really not be located in the code.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomMcsClaimGeneratorTest.java [hunk 14]
```diff
@@ -53,88 +53,265 @@ public void randomClaim() throws InvalidProtocolBufferException {
...
-            + "  }, {\n"
-            + "    \"idrDtlStatusEnum\": \"DETAIL_STATUS_SENT\",\n"
-            + "    \"idrDtlFromDate\": \"2021-01-19\",\n"
-            + "    \"idrDtlToDate\": \"2021-04-13\",\n"
-            + "    \"idrProcCode\": \"107gz\",\n"
-            + "    \"idrModOne\": \"n\",\n"
-            + "    \"idrModTwo\": \"x\",\n"
-            + "    \"idrModThree\": \"v\",\n"
-            + "    \"idrModFour\": \"vw\",\n"
+            + "    \"idrDtlStatusEnum\": \"DETAIL_STATUS_PENDING\",\n"
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomMcsClaimGeneratorTest.java:59]: Are these just randomly generated and copied into these tests? Seems a lot easier just to call the randomizer code, turn it into json programmatically, and pass that to the test. This _looks_ kinda manual?

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java [hunk 15]
```diff
@@ -0,0 +1,384 @@
...
+/**
+ * The claim transformers {@link FissClaimTransformer} and {@link McsClaimTransformer} combined
+ * populate 11 different JPA entity classes with data from corresponding gRPC API message objects.
+ * The tests for these transformations need to apply specific checks for every field based on the
+ * type of field being converted. These tests have a lot of commonality but work on different
+ * protobuf and JPA objects that aren't connected through inheritance or any interfaces.
+ *
+ * <p>This abstract class implements the field checks using a set of abstract methods overridden by
+ * entity specific derived classes to instantiate message and entity objects of the correct type.
+ *
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java:24]: Nice doc!

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java [hunk 16]
```diff
@@ -0,0 +1,384 @@
...
+          Function<TTestEntity, BigDecimal> getter,
+          String fieldLabel) {
+    final BiConsumer<TClaimBuilder, String> wrappedSetter =
+        (claimBuilder, value) -> setter.accept(getTestEntityBuilder(claimBuilder), value);
+    final Function<TClaimEntity, BigDecimal> wrappedGetter =
+        claim -> getter.apply(getTestEntity(claim));
+    verifyFieldTransformationSucceeds(
+        claimBuilder -> wrappedSetter.accept(claimBuilder, "1234.50"),
+        wrappedGetter,
+        new BigDecimal("1234.50"));
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java:166]: 🍰  Could maybe also check that a number with no decimal is properly converted? Something like 56 and/or 56.5. I'd assume the converter should properly be setting the correct scaling to 2 decimals even if not provided.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformerTest.java [hunk 17]
```diff
@@ -412,6 +418,48 @@ public void benezPayer() {
         claim.getPayers(), transformed.getPayers(), PreAdjFissPayer::getPriority);
   }
 
+  @Test
+  public void auditTrail() {
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformerTest.java:422]: This test name and the others below could be a bit more descriptive. (And documented)

**Review by lsmitchell — COMMENTED** [review]
I like all the detailed field transformer testing! Some of it is a little hard to follow due to being a series of calls and abstract overrides; test naming and documentation might help a little.

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissPayer.java (hunk 10 above)
Comment (brianburton) [reply apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissPayer.java:112]: Good point.

Since we are using Hibernate and want to keep the implementation straight-forward this seemed like a good approach.  If we used multiple subclasses with different fields in each subclass we'd wind up having to do `instanceof` calls in code that queried the table.

Maybe we could map those fields to sub-entities and have hibernate populate them into fields like `getBeneZPayer()` and `getInsuredPayer()`.  That would be a whole new PR though since these fields pre-exist 0.5.

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsAdjustment.java (hunk 11 above)
Comment (brianburton) [reply apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsAdjustment.java:46]: I grabbed those comments from the RDA API proto files.  Not all of the fields have comments though.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomMcsClaimGeneratorTest.java (hunk 14 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomMcsClaimGeneratorTest.java:59]: This is testing the random claim generator.  That expected JSON is taken from a run of the generator with a known seed value and compared to a run during the test to see if the output has changed from the known good one.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformerTest.java (hunk 17 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformerTest.java:422]: Will do.

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissPayer.java (hunk 10 above)
Comment (lsmitchell) [reply apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjFissPayer.java:112]: Yup, I'd agree that would be work for another PR, but something to think about for the future 👍

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomMcsClaimGeneratorTest.java (hunk 14 above)
Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/server/RandomMcsClaimGeneratorTest.java:59]: Ah I see, thanks for explaining!

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java (hunk 16 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java:166]: I've allowed the BigDecimal constructor to determine the scale based on the string we receive from the RDA API.  The actual precision when written to the database will be set by the database column definition.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformerTest.java (hunk 17 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/FissClaimTransformerTest.java]: Done.

File: apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsDetail.java (hunk 12 above)
Comment (brianburton) [reply apps/bfd-model/bfd-model-rda/src/main/java/gov/cms/bfd/model/rda/PreAdjMcsDetail.java:117]: I grabbed those comments from the RDA API proto files. Not all of the fields have comments though.

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java (hunk 16 above)
Comment (lsmitchell) [reply apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/test/java/gov/cms/bfd/pipeline/rda/grpc/source/ClaimTransformerFieldTester.java:166]: Got it. Thanks for explaining!

**Review by lsmitchell — DISMISSED** [review]

**Review by christrotteradhoc — APPROVED** [review]
